### PR TITLE
feat(github): Add New issue Form

### LINKED ISSUE #519: Add GitHub Issue Template Schema Form
**Description:**
To streamline the process of creating issues and ensure consistency, we need to add a GitHub issue template schema form. This will help contributors easily provide all the necessary details when reporting bugs, suggesting features, or requesting documentation.

**Requirements:**
1. Create an issue template that includes fields for:
   - Issue Type (Bug, Feature Request,Documentation)
   - Description
   - Steps to Reproduce (for Bug reports)
   - Expected Outcome
   - Additional Context or Screenshots
2. Ensure the template is easy to navigate and fills out the required fields.

**Proposed Changes:**
1. Implement the issue template in `.github/ISSUE_TEMPLATE` folder.
2. Test the template to verify the form fields populate correctly.

**Before and After:**

*Before:*
Screenshot 2024-10-09 at 7 03 35 PM
  
*After:*
Screenshot 2024-10-09 at 7 02 52 PM

**Additional Information:**
If you have any suggestions for extra fields or modifications, please mention them in the comments.

PS: I am willing to submit the Pull Request

### PR DESCRIPTION
**Pull request by utsavdotdev**
### PR Description
Introduces the new issue template using YML schema form. 
Fix #519 

### Some notes

- [x] Please make sure that you placed the files in an appropriate folder
- [x] And that the files have indicative names.

- [x] Please note that Classiq runs automatic code linting, which may minorly alter some files.
  - [x] If you're familiar with `pre-commit`, you may run `pre-commit install`, and then at each commit, your files will be altered in a similar way

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        description: Please copy and paste any relevant log output. This will be automatically formatted into code, so no need for backticks.
+        render: shell
+
+    - type: checkboxes
+      id: no-duplicate-issues
+      attributes:
+        label: "👀 Have you spent some time to check if this bug has been raised before?"
+        options:
+          - label: "I checked and didn't find similar issue"
+            required: true
```
Comment (TaliCohn) [inline .github/ISSUE_TEMPLATE/bug_report.yml:68]: This doesn't have to be mandatory. We are happy to get multiple reports about the same bug.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+
+    - type: checkboxes
+      id: terms
+      attributes:
+        label: 🧑‍⚖️ Code of Conduct
+        description: By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/Classiq/classiq-library/blob/main/CODE_OF_CONDUCT.md)
+        options:
+          - label: I agree to follow this project's Code of Conduct
+            required: true
```
Comment (TaliCohn) [inline .github/ISSUE_TEMPLATE/bug_report.yml:89]: This is unnecessary, can be deleted.

File: .github/ISSUE_TEMPLATE/docs_improvement.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            description: "Any additional context or information that might be helpful."
+            placeholder: "e.g., Links to related documentation, screenshots, etc."
+
+    - type: checkboxes
+      id: no-duplicate-issues
+      attributes:
+         label: "👀 Have you spent some time to check if this issue has been raised before?"
+         options:
+          - label: "I checked and didn't find similar issue"
+            required: true
```
Comment (TaliCohn) [inline .github/ISSUE_TEMPLATE/docs_improvement.yml:37]: This doesn't have to be mandatory.

File: .github/ISSUE_TEMPLATE/docs_improvement.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        required: true
+
+    - type: checkboxes
+      id: terms
+      attributes:
+        label: 🧑‍⚖️ Code of Conduct
+        description: By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/Classiq/classiq-library/blob/main/CODE_OF_CONDUCT.md)
+        options:
+          - label: I agree to follow this project's Code of Conduct
+            required: true
```
Comment (TaliCohn) [inline .github/ISSUE_TEMPLATE/docs_improvement.yml:57]: This is unnecessary, can be deleted.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      description: "A clear and concise description of any alternative solutions or additional solutions you've considered."
+      placeholder: "I tried, ..."
+
+  - type: checkboxes
+    id: no-duplicate-issues
+    attributes:
+      label: "👀 Have you spent some time to check if this feature request has been raised before?"
+      options:
+        - label: "I checked and didn't find similar issue"
+          required: true
```
Comment (TaliCohn) [inline .github/ISSUE_TEMPLATE/feature_request.yml:52]: This doesn't have to be mandatory.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      required: true
+
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: 🧑‍⚖️ Code of Conduct
+      description: By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/Classiq/classiq-library/blob/main/CODE_OF_CONDUCT.md)
+      options:
+          - label: I agree to follow this project's Code of Conduct
+            required: true
```
Comment (TaliCohn) [inline .github/ISSUE_TEMPLATE/feature_request.yml:72]: This is unnecessary, can be deleted.

**Review by TaliCohn — CHANGES_REQUESTED** [review]

**Comment by utsavdotdev** [comment]
Got it. I will push the required changes.

**Comment by utsavdotdev** [comment]
@TaliCohn , I have made the required changes. Can you review it?

**Review by TaliCohn — CHANGES_REQUESTED** [review]
Please delete this from all the forms. As I mentioned, we do not mind having duplicated issues and we don't wont people not to report because of this section. 

 attributes:
   label: "👀 Have you spent some time to check if this feature request has been raised before?"
      options:
        - label: "I checked and didn't find similar issue"
          required: false

**Comment by utsavdotdev** [comment]
ok

**Comment by utsavdotdev** [comment]
@TaliCohn , Added the Requested changes

**Review by TaliCohn — APPROVED** [review]

**Comment by NadavClassiq** [comment]
Hello @utsavdotdev!
Could you please rebase your branch on top of the main branch?

**Comment by utsavdotdev** [comment]
@NadavClassiq Done

**Comment by NadavClassiq** [comment]
Hello again, @utsavdotdev !

Thanks for the quick response. It looks like GitHub flagged an issue with this PR, possibly due to a `git merge` instead of a `git rebase` (which is our methodology here in this repo).

To ensure you’re credited as the contributor, could you open a new PR from an updated branch? Please update your fork, then either download the files and start a new branch or cherry-pick the commit from this branch.

Let us know if you need any help!

**Pull request closed without merging**